### PR TITLE
adds better grass sprites

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

adds better grass sprites, ported from eris
converts fairygrass into those sprites too
makes "plating" grass and jungle grass smooth with cool stuff on the edges
adds a sandy dirt tile, removes the old "sand" icon which was just puke colored noise
image

## Why It's Good For The Game

a sprite that isnt just green colored noise is better

## Changelog
:cl: Fikou, sprites from CEV Eris
imageadd: adds better grass sprites
/:cl:

### CODE REVIEW

File: code/game/turfs/open/floor/fancy_floor.dm [hunk 1]
```diff
@@ -122,6 +125,9 @@
 	clawfootstep = FOOTSTEP_SAND
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
 
+/turf/open/floor/grass/snow/spawniconchange()
```
Comment (tralezab) [inline code/game/turfs/open/floor/fancy_floor.dm:128]: try to detect the real issue here

File: code/game/turfs/open/floor/fancy_floor.dm [hunk 2]
```diff
@@ -168,6 +174,8 @@
 		icon_state = "basalt[rand(0, 12)]"
 		set_basalt_light(src)
 
+/turf/open/floor/grass/fakebasalt/spawniconchange()
```
Comment (tralezab) [inline code/game/turfs/open/floor/fancy_floor.dm:177]: ditto

File: code/game/turfs/open/floor/fancy_floor.dm (hunk 1 above)
Comment (Fikou) [reply code/game/turfs/open/floor/fancy_floor.dm:128]: by yours power combined, i am captain planet

**Comment by Qustinnus** [comment]
Post some pictures with the actual changes on junglegen.

**Comment by Qustinnus** [comment]
on a seperate note, this should be in an element that handles both the transforming of the turf and has some kind of layering system to make it less of a PITA to have multiple of these types of turfs. if you need help with that ping me but if we dont do this it'll probably bite us in the ass later.

**Review by Twaticus — APPROVED** [review]

File: code/game/turfs/open/floor/plating/misc_plating.dm [hunk 3]
```diff
@@ -271,13 +271,37 @@
 /turf/open/floor/plating/grass
 	name = "grass"
 	desc = "A patch of grass."
-	icon_state = "grass"
-	broken_states = list("sand")
+	icon_state = "grass0"
+	var/smooth_icon = 'icons/turf/floors/grass.dmi'
```
Comment (Rohesie) [inline code/game/turfs/open/floor/plating/misc_plating.dm:275]: But new variables under the redefined ones. In this case, put this variable under the `layer` one.

File: code/game/turfs/open/floor/plating/misc_plating.dm [hunk 4]
```diff
@@ -271,13 +271,37 @@
...
+	canSmoothWith = list(SMOOTH_GROUP_CLOSED_TURFS,SMOOTH_GROUP_FLOOR_GRASS)
+	layer = HIGH_TURF_LAYER
+
+/turf/open/floor/plating/grass/Initialize()
+	if(smoothing_flags)
+		var/matrix/M = new
+		M.Translate(-9, -9)
+		transform = M
+		icon = smooth_icon
+	. = ..()
```
Comment (Rohesie) [inline code/game/turfs/open/floor/plating/misc_plating.dm:292]:
```suggestion
	. = ..()
	if(smoothing_flags)
		var/matrix/translation = new
		translation.Translate(-9, -9)
		transform = translation
		icon = smooth_icon
```
* Always call parent first on `Initialize()` when you can get away with it.
* Use better variable names.

**Pull request merged**